manifests/certonly.pp

```diff
@@ -112,21 +117,31 @@
 
   $execution_environment = [ "VENV_PATH=${letsencrypt::venv_path}", ] + $environment
   $verify_domains = join(unique($domains), ' ')
-  exec { "letsencrypt certonly ${title}":
-    command     => $command,
-    path        => $facts['path'],
-    environment => $execution_environment,
-    unless      => "/usr/local/sbin/letsencrypt-domain-validation ${live_path} ${verify_domains}",
-    provider    => 'shell',
-    require     => [
-      Class['letsencrypt'],
-      File['/usr/local/sbin/letsencrypt-domain-validation'],
-    ],
+  if $ensure == 'present' {
+    exec { "letsencrypt certonly ${title}":
+      command     => $command,
+      path        => $facts['path'],
+      environment => $execution_environment,
+      unless      => "/usr/local/sbin/letsencrypt-domain-validation ${live_path} ${verify_domains}",
+      provider    => 'shell',
+      require     => [
+        Class['letsencrypt'],
+        File['/usr/local/sbin/letsencrypt-domain-validation'],
+      ],
+    }
+  } else {
+    file { "${config_dir}/live/${live_path_domain}/":
```

Comment:
Did some testing to see what this'd do out of curiosity/concern. I'd personally look at running an exec to have certbot itself delete the certificate as rm -rf'ing the directory isn't quite sufficient to clean it all up. There are still files left behind in /etc/letsencrypt/renewal and /etc/letsecrypt/archive.

```
/etc/letsencrypt/renewal/certone.test.rawr.no.conf
/etc/letsencrypt/archive/certone.test.rawr.no
```

Certbot is unhappy about them as well:

```
$ certbot certificates
...
Renewal configuration file /etc/letsencrypt/renewal/certone.test.rawr.no.conf produced an unexpected error: expected /etc/letsencrypt/live/certone.test.rawr.no/cert.pem to be a symlink. Skipping.
```

Creating a new certificate again with the same name without cleaning up these files makes it append a -0001 to it, leaving the old files behind and generating a certificate name the module can't clean up by itself.

What about something like this?
```puppet
exec { "letsencrypt cleanup ${title}":
  command => "${letsencrypt_command} delete --cert-name ${title}",
  path    => $facts['path'],
  onlyif  => "test -d '${config_dir}/live/${live_path_domain}'",
}
```

Reply:
That's a much better idea. I'll look into implementing that over the coming week.